crates/matrix-sdk-ui/src/timeline/mod.rs

```diff
@@ -568,6 +594,38 @@ impl Timeline {
         SendAttachment::new(self, path.into(), mime_type, config)
     }
 
+    /// Redact an event given its [`TimelineEventItemId`] and an optional
+    /// reason.
+    ///
+    /// See [`Self::redact`] for more info.
+    pub async fn redact_by_id(
```

Comment:
Can we extract this new method in a separate PR?

Reply:
I'm not sure about this one though, because at the FFI layer we no longer have a `sdk_ui::EventTimelineItem` we can pass to the existing `fn redact`. 

We *could* get it by passing an `ffi::EventTimelineItem`, checking its event and transaction ids, using them to look for the event in the timeline and then passing it down to the existing function. If we're alright with that implementation this one can be delayed into a next one, but it means we'll have to adapt the client's API twice instead of just once.

If we don't do this separate change we'd end up doing this:

1. In the client, call `val event = timeline.getTimelineItemByEventId(eventId)` or `val event = timeline.getTimelineItemByTransactionId(transactionId)`.
2. This looks for the event in the timeline, transforms it ands passes it back through the FFI layer.
3. We then call `timeline.redact(event)` in the client.
4. This will get the event id or transaction id from the FFI record, search for the sdk_ui event again, and *then* do the operation.

This is not wrong, and it would still use the existing API, but it's definitely not a good implementation IMO.

Reply:
I think he meant that we would first add this method in a separate PR, without the removal of the stuff from the FFI layer, then this PR would depend on the PR introducing this method.

But since this PR has been open for quite a while and this is a fairly minor point, let's not split it out at this point in time anymore.